inst/assets/BS4/pkgdown.js

```diff
@@ -64,5 +64,85 @@
 
     });
   }
+
+    /* Search marking --------------------------*/
+    var url = new URL(window.location.href);
+    var toMark = url.searchParams.get("q");
+    var mark = new Mark("div.col-md-9");
+    if (toMark) {
+      mark.mark(toMark, {
+        accuracy: {
+          value: "complementary",
+          limiters: [",", ".", ":", "/"],
+        }
+      });
+    }
+
+  /* Search --------------------------*/
+    // Initialise search index on focus
+  var fuse;
+  $("#search-input").focus(async function(e) {
+    if (fuse) {
+      return;
+    }
+
+    $(e.target).addClass("loading");
+    var response = await fetch($("#search-input").data("search-index"));
+    var data = await response.json();
+
+    var options = {
+      keys: ["heading", "text", "code"],
+      ignoreLocation: true,
+      threshold: 0.1,
+      includeMatches: true,
+      includeScore: true,
+    };
+    fuse = new Fuse(data, options);
+
+    $(e.target).removeClass("loading");
+  });
+
+  // Use algolia autocomplete
+  var options = {
+    autoselect: true,
+    debug: true,
+    hint: false,
+    minLength: 2,
+  };
+  var q;
+async function searchFuse(query, callback) {
+  await fuse;
+
+  var items;
+  if (!fuse) {
+    items = [];
+  } else {
+    q = query;
+    var results = fuse.search(query, { limit: 20 });
+    items = results
+      .filter((x) => x.score <= 0.75)
+      .map((x) => x.item);
+  }
+  callback(items);
+}
+  $("#search-input").autocomplete(options, [
+    {
+      name: "content",
+      source: searchFuse,
+      templates: {
+        suggestion: (s) => {
+          if (s.title == s.heading) {
+            return `<div class="search-details">${s.title}</div>`;
+          } else {
+            return `${s.title}	<br><div class="search-details"> ${s.heading}</div>`;
```

Comment:
I like the use of different sizes and colors that Algolia has, so I did some work around that, without being sure it's very pretty yet.